### PR TITLE
Added api to access table information as json api

### PR DESCRIPTION
**Pull request by afshawnlotfi**
I wanted to do some automated scripts that fetched table information and generated an AWS Cloudformation script from the table information from this project. All this PR does is exposes the json used in the "/" endpoint as raw json in the endpoint "/api/list-all". I also added support for devcontainers in Vscode so a person doesn't have to worry about setting their dev environment up they just run the project in a Remote docker container. More info here [https://code.visualstudio.com/docs/remote/remote-overview](https://code.visualstudio.com/docs/remote/remote-overview).

### CODE REVIEW

File: lib/backend.js [hunk 1]
```diff
@@ -137,28 +137,35 @@ exports.createServer = (dynamodb, docClient) => {
...
+  }
 
+  app.get('/', asyncMiddleware((req, res) => {
     return listAllTables(null, [])
       .then(data => {
         res.render('tables', { data })
       })
   }))
 
+  app.get('/api/list-all', (req, res) => {
```
Comment (rchl) [inline lib/backend.js]: I'd prefer, per REST convections, use path like `/api/tables` here.

**Comment by rchl** [comment]
I've integrated your API endpoint changes.

As for the devcontainers configuration, I have some issues with it but I didn't want to block the endpoint changes so feel free to submit PR for devcontainer changes separately.

Iinitial thoughs on devcontainer changes:
 - VSCode shows listing of whole server rather than just the project directory
 - There is no configuration for starting the dynamodb_admin app

Both of those are handled in the example app listed in devcontainers documentation. I've tried with this project: https://github.com/Microsoft/vscode-remote-try-node

**Comment by afshawnlotfi** [comment]
I purposely didn't auto launch dynamodb_admin app becuase this isn't a production docker image, the developer can launch and kill the task whenever they please. But you can do whatever. Already forked the project and everything on my end is working so you don't have to merge that part in. As for the listing of the whole image directories I also did that purposely because I wanted to store database entries at /data/db but you can disable it.

**Comment by rchl** [comment]
I didn't mean auto lunch. Example project includes launch configuration so developer can just press F5 to start it.

**Pull request closed without merging**